### PR TITLE
[All] remove method applyTestCreateFunction in TopologyDataHandler 

### PR DESCRIPTION
**Pull request by epernod**
This PR will be merged in the topologyChanges_POC branch

this method was never called and is not useful.
Add Disabled message.

This need to be merged before #2114

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

File: SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h [hunk 1]
```diff
@@ -139,17 +139,19 @@ class TopologyDataHandler : public sofa::core::topology::TopologyHandler
...
+
 
     // update the default value used during creation
     void setDefaultValue(const value_type& v) {
         m_defaultValue = v;
     }
 
+
+    ////////////////////////////////////// DEPRECATED ///////////////////////////////////////////
+    // TODO change message before PR
```
Comment (alxbilger) [inline SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h]: Is this TODO still necessary?

File: SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h [hunk 2]
```diff
@@ -139,17 +139,19 @@ class TopologyDataHandler : public sofa::core::topology::TopologyHandler
...
+    SOFA_ATTRIBUTE_DISABLED("v21.06 (PR#2171)", "v21.06 (PR#2171)", "This method has been removed as it has never been used and is not needed.")
```
Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h:152]: Maybe add a fallback method (applyCreateFunction?) in the help message.

File: SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h (hunk 1 above)
Comment (guparan) [reply SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h]: Obviously no

Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h]:
```suggestion
```

File: SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h (hunk 2 above)
Comment (guparan) [inline SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyDataHandler.h]: Maybe add a fallback method (applyCreateFunction?) in the help message.

**Review by guparan — CHANGES_REQUESTED** [review]

**Review by alxbilger — APPROVED** [review]

**Pull request merged**